hack/lib/golang.sh

```diff
@@ -479,7 +479,7 @@ EOF
   local go_version
   IFS=" " read -ra go_version <<< "$(GOFLAGS='' go version)"
   local minimum_go_version
-  minimum_go_version=go1.15.0
+  minimum_go_version=go1.16.0
```

Comment:
```suggestion
  minimum_go_version=go1.16
```

Reply:
the comparison handles go1.16 satisfying go1.16.0, leaving it fully specified seems preferable to me

Reply:
1.16.0 is not a valid go version technically, because go. https://github.com/golang/go/issues/27255#issuecomment-424812737

Reply:
From [Slack](https://kubernetes.slack.com/archives/CHGFYJVAN/p1614716625058500?thread_ts=1614712814.055800&cid=CHGFYJVAN):

justaugustus (he/him)  5 minutes ago
> @liggitt -- hmmm, the review was more about the fact that:

```console
$ go version
go version go1.16 linux/amd64
```

justaugustus (he/him)  5 minutes ago
> i.e., .0 is elided in the upstream official version

liggitt  3 minutes ago

> understood, but the `echo -e "${minimum_go_version}\n${go_version[2]}" | sort -s -t. -k 1,1 -k 2,2n -k 3,3n | head -n1` comparison allows an actual go version of go1.16 to pass the minimum_go_version=go1.16.0 requirement